share-coordinator/src/main/java/org/apache/kafka/coordinator/share/ShareCoordinatorShard.java

```diff
@@ -274,79 +270,89 @@ public CoordinatorResult<WriteShareGroupStateResponseData, CoordinatorRecord> wr
             return error.get();
         }
 
-        String groupId = request.groupId();
         WriteShareGroupStateRequestData.WriteStateData topicData = request.topics().get(0);
         WriteShareGroupStateRequestData.PartitionData partitionData = topicData.partitions().get(0);
+        SharePartitionKey key = SharePartitionKey.getInstance(request.groupId(), topicData.topicId(), partitionData.partition());
+
+        CoordinatorRecord record = getRecord(partitionData, key);
+
+        // build the response based on record
```

Comment:
Hmm, the response doesn't seem to be built on record.

Reply:
The response for this RPC is just a null error code - hence no actual data from the record is required. This is a write operation.

Reply:
The null checks are were added in response to https://github.com/apache/kafka/pull/17149#discussion_r1783239363

`constainsKey` will still be true if the key maps to a null.